### PR TITLE
SCIP & SCIP-TypeScript Blogposts

### LINKED ISSUE #5313: Blog post: announce LSIF Typed

### PR DESCRIPTION
**Pull request by zlonko**
This closes #5313 and #5311. It creates the blogpost markdown and thumbnails for the SCIP blogpost and the SCIP-TypeScript blogpost.

- [SCIP blogpost](https://docs.google.com/document/d/11ENvw9axbxX649xWTIMXn4Ybj_OwIfJVcpCXVA6KdDA/edit#)
- [SCIP-TypeScript blogpost](https://docs.google.com/document/d/1e5-jDZJWmaVPl8ujni_2LTh2xfxrbC5YYob7cVxlzeI/edit#)

### Changelog

- Blogpost copy for `/blog/announcing-scip` and `/blog/announcing-scip-typescript`
- Embedded tweet on announcing-scip
- Embedded figures on announcing-scip-typescript
- Associated hero and social media images

### Notes
- Some links were missing in the Google Docs. For those labeled "announcements", I linked to the corresponding blogpost. On `announcing-scip`, there is an intended link to a "scip-java" announcement. Where do we want this link to lead?
- All of the images are hosted on GCP

### Test

1. Ensure prettier has standardized the proposed changes.
2. Please check that copy is transcribed accurately from the Google Docs 
3. Please ensure images render correctly and links are active

### CODE REVIEW

File: content/blogposts/2022/announcing-scip.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+We are excited to announce SCIP, a new indexing format that we are using at Sourcegraph to index programming languages to power code navigation features such as “Go to definition” and “Find references.” We have been using another format called LSIF (Language Server Index Format) for this purpose and in this blog post we want to share the challenges we faced by using LSIF and how that led us to create SCIP. Code navigation is a critical component of the user experience on Sourcegraph and the underlying code navigation data, the main topic of this blog post, is also a building block for a suite of other products under Sourcegraph’s offering as a code intelligence platform.
+
+This blog post is technical and primarily aimed at authors of developer tools that produce or consume metadata about programming languages. If you’re a Sourcegraph user and want to jump straight to trying out precise code navigation with scip-typescript (for TypeScript and JavaScript) or scip-java (for Java, Scala, and Kotlin), you can read more in their respective announcements, [here](announcing-scip-typescript) and [here](LINK).
```
Comment (olafurpg) [inline content/blogposts/2022/announcing-scip.md:18]:
```suggestion
This blog post is technical and primarily aimed at authors of developer tools that produce or consume metadata about programming languages. If you’re a Sourcegraph user and want to jump straight to trying out precise code navigation with scip-typescript (for TypeScript and JavaScript) or scip-java (for Java, Scala, and Kotlin), you can read more in their respective announcements, [here](announcing-scip-typescript) and [here](https://github.com/sourcegraph/scip-java/releases/tag/v0.8.0).
```

**Review by olafurpg — APPROVED** [review]
Thank you @zlonko for porting the draft to markdown 🙏 I've released scip-java v0.8 and suggested changes to link to the release

File: content/blogposts/2022/announcing-scip.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+## Compatibility with LSIF
+
+While we are excited to grow our usage of SCIP, we care deeply about compatibility with the existing LSIF ecosystem and will continue to accept LSIF uploads on Sourcegraph. Any tool that consumes LSIF can also consume SCIP by running our [`scip`](https://github.com/sourcegraph/scip) command-line tool that converts a SCIP payload into an LSIF v0.4.3 payload. In fact, we use this compatibility layer ourselves at the moment when uploading SCIP indexes to Sourcegraph because our backend still only processes LSIF.
+
+## Index TypeScript and Java with SCIP today
+
+To pair with this announcement, we are releasing two new SCIP indexers:
+
+- **scip-typescript**: TypeScript and JavaScript indexer built on top of the TypeScript typechecker. Learn more about using this indexer here ([link](announcing-scip-typescript)).
+- **scip-java**: Java, Scala, and Kotlin indexer built on top of compiler plugins for each respective language. Learn more about using this indexer here ([link](LINK)).
```
[the remark this answers is not in the document]
Comment (zlonko) [reply content/blogposts/2022/announcing-scip.md:79]: Right on! Thank you, Olaf. I accepted the suggested change so we should be good to go!

**Comment by zlonko** [comment]
Hi @olafurpg! Accepting your changes dismissed your approval in GH. Sorry to tag you again!

**Review by st0nebraker — APPROVED** [review]
LGTM! Non-blocking question/suggestion: I know the blog section of our repo is inconsistent with assets. But I believe smaller assets should be in our repo while larger ones are in GCP, [Handbook: Making blogposts](https://handbook.sourcegraph.com/departments/marketing/blog/#uploading-images). We should also update the file paths for assets since we don't have the `website` directory anymore 😬 . Is that accurate @bretthayes ? This is non-blocking if we need to get these out, though!

File: content/blogposts/2022/announcing-scip-typescript.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+This is a good example of the kind of static analysis that scip-typescript is able to perform. We are excited to extend the functionality of scip-typescript to include related navigation actions like “Go to type definition.”
+
+## Migrating from lsif-node to scip-typescript
+
+Before creating scip-typescript, we used another TypeScript indexer called [lsif-node](https://github.com/sourcegraph/lsif-node). We recommend migrating to scip-typescript if you are using lsif-node.
+
+Follow the steps below to migrate from lsif-node to scip-typescript:
+
+Replace usages of the `lsif-tsc -p ARGUMENTS` command with `scip-typescript index ARGUMENTS`.
```
[the remark this answers is not in the document]
Comment (zlonko) [reply content/blogposts/2022/announcing-scip-typescript.md]: Great catch!

File: content/blogposts/2022/announcing-scip-typescript.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  <img src="https://storage.googleapis.com/sourcegraph-assets/blog/announcing-scip-typescript/trigger-find-references.png" alt="Trigger Find references on action() method" className="no-shadow"></img>
+</figure>
+
+Observe that the results come from both GitHub repositories and npm packages. The code from npm packages is the same source code that’s typically installed under the `node_modules/` directory.
+
+Read more in our docs, [here](https://docs.sourcegraph.com/integration/npm), on how to set up the same npm package support on a self-hosted Sourcegraph instance.
+
+## Find implementations
+
+Use the new “Find implementations” button to navigate from an abstract class, interface, interface property, or abstract class method to their concrete implementations. For example, trigger “Find implementations” on the [`QuickPickItem.label`](https://sourcegraph.com/npm/types/vscode@b309120c719af01453d6df4a7f82902c22b1afb3/-/blob/index.d.ts?L1678:9&subtree=true#tab=implementations_typescript) property from the `@types/vscode` npm package.
```
[the remark this answers is not in the document]
Comment (zlonko) [reply content/blogposts/2022/announcing-scip-typescript.md:131]: Sure thing. Let us know anytime we can support updating this post in the future.

**Comment by zlonko** [comment]
Made a small change to the links on `announcing-scip` per feedback from Alex I: d529a4c

**Review by bretthayes — APPROVED** [review]

**Review by olafurpg — APPROVED** [review]
LGTM 👍

**Pull request merged**